src/dune_rules/lib_flags.ml

```diff
@@ -92,19 +92,26 @@ module L = struct
        |> List.rev)
   ;;
 
-  let remove_stdlib dirs libs =
+  let to_flags dirs =
+    Command.Args.S
+      (Path.Map.foldi dirs ~init:[] ~f:(fun dir direct acc ->
+         Command.Args.Path dir :: A (if direct then "-I" else "-H") :: acc)
+       |> List.rev)
+  ;;
+
+  let remove_stdlib dirs libs remove =
     match libs with
     | [] -> dirs
-    | lib :: _ -> Path.Set.remove dirs (Lib.lib_config lib).stdlib_dir
+    | lib :: _ -> remove dirs (Lib.lib_config lib).stdlib_dir
   ;;
 
   type mode =
     { lib_mode : Lib_mode.t
     ; melange_emit : bool
     }
 
-  let include_paths =
-    let add_public_dir ~visible_cmi obj_dir acc mode =
+  let include_paths_gen =
```

Comment:
`~init`, `~add` and `~remove` should be arguments of `include_paths_gen` directly.